lighthouse-core/audits/preload-lcp-image.js

```diff
@@ -110,10 +110,64 @@ class PreloadLCPImageAudit extends Audit {
    * @return {{wastedMs: number, results: Array<{url: string, wastedMs: number}>}}
    */
   static computeWasteWithGraph(lcpNode, graph, simulator) {
-    const simulation = simulator.simulate(graph, {flexibleOrdering: true});
-    // For now, we are simply using the duration of the LCP image request as the wastedMS value
-    const lcpTiming = simulation.nodeTimings.get(lcpNode);
-    const wastedMs = lcpTiming && lcpTiming.duration || 0;
+    const simulationBeforeChanges = simulator.simulate(graph, {flexibleOrdering: true});
+    const modifiedGraph = graph.cloneWithRelationships();
+
+    // Store the IDs of the LCP Node's dependencies for later
+    /** @type {Set<string>} */
+    const dependenciesIds = new Set();
+    for (const node of lcpNode.getDependencies()) {
+      dependenciesIds.add(node.id);
+    }
+
+    /** @type {LH.Gatherer.Simulation.GraphNode|null} */
+    let modifiedLCPNode = null;
+    /** @type {LH.Gatherer.Simulation.GraphNode|null} */
+    let mainDocumentNode = null;
+    modifiedGraph.traverse(node => {
+      if (node.type !== 'network') return;
+
+      const networkNode = /** @type {LH.Gatherer.Simulation.GraphNetworkNode} */ (node);
+      if (node.isMainDocument()) {
+        mainDocumentNode = networkNode;
+      } else if (networkNode.record.url === lcpNode.record.url) {
```

Comment:
maybe just eaiser to compare ids here too so we're consistently comparing ids everywhere?